test/index.js

```diff
@@ -1277,7 +1278,7 @@ describe('fixtures', function () {
 
                     compare(node, trees[mapping[key]], false, initialClean);
 
-                    markdown = remark.stringify(node, stringify);
+                    markdown = remark.stringify(clone(node), stringify);
```

Comment:
This is needed for some [stringify tests](https://github.com/wooorm/remark/blob/3802139667b336bdcbcfdc7f7859391456ab0d37/test/index.js#L1285) because node mergers currently [modify AST nodes](https://github.com/wooorm/remark/blob/3802139667b336bdcbcfdc7f7859391456ab0d37/lib/utilities.js#L257) during compilation, and because of that it can sometimes become invalid. I would suggest returning new nodes and avoid mutations in compiler instead. What do you think?